tests/__init__.py

```diff
@@ -152,6 +152,14 @@ def server(self):
     def client(self):
         yield RatbagemuClient()
 
+    @pytest.fixture()
+    def id(self, client):
+        id = self.add_device(client)
+
+        yield id
+
+        client.delete_device(id)
```

Comment:
still not sure what the `pytest.fixture()` bit does so please expand the commit messages to explay **why** a change is needed

Reply:
What do you mean? `pytest.fixture` will mark the function defined below as a fixture. We then can do something like this.

```python
def test_feature_something(id):
	# Test the feature here...
```

Seeing this pytest will call the `id()` function, which will create a device and return the id, call `test_feature_something()` with that id and finally come back to `id()` and delete the device.

I am not sure what is confusing here. This is basic pytest API. I think it would be acceptable to expect developers to know a bit of pytest when looking at the tests history.

Reply:
the big problem with libratbag is that we work on it so infrequently that much of the knowledge gets mentally paged out, commit messages help to page the content back in with a bit of context. So a quick reminder of why you're adding this (like you did just above) helps not having to sit down and spend 20 min re-reading the code for it all to make sense again.

There's plenty of grey area of things you can expect and can't expect, so there's no right answer. I found some time to read up on fixtures, so at least I understand that bit now - however - I don't see where you're using this fixture now which makes it more confusing. If this is the first of a few commits that use it, ok, but (at least with the github interface) it's nontrivial to figure out where this is being used.

Reply:
Okay, makes sense :)

It is used in the HID++ 2.0 tests. I don't use it in the server API tests as there we need to create specific devices with specific initial states for each test, so we can't do it automatically.